lib/miq_automation_engine/service_models/miq_ae_service_container_deployment.rb

```diff
@@ -0,0 +1,194 @@
+require 'net/scp'
+
+module MiqAeMethodService
+  class MiqAeServiceContainerDeployment < MiqAeServiceModelBase
+    expose :container_deployment_nodes, :association => true
+    expose :deployed_ems, :association => true
+    expose :deployed_on_ems, :association => true
+    expose :automation_task, :association => true
+    expose :roles_addresses
+    expose :container_nodes_by_role
+    expose :ssh_auth
+    expose :rhsm_auth
+    SUBSCRIPTION_REPOS = ["rhel-7-server-rh-common-rpms", "rhel-7-server-rpms", "rhel-7-server-extras-rpms", "rhel-7-server-ose-3.2-rpms"].freeze
+    ANSIBLE_LOG = "/tmp/ansible.log".freeze
+    ANSIBLE_ERROR_LOG = "/tmp/openshift-ansible.log".freeze
+    INVENTORIES_PATH = "/usr/share/ansible/openshift-ansible/".freeze
+    SSH_AGENT_PATH = "/tmp/ssh_manageiq/".freeze
+
+    def assign_container_deployment_node(vm_id, role)
```

Comment:
@alongoldboim Shouldn't these functions be defined in the app/models and we just expose them from the service models. Our long term goal is to push things into the app/models and expose the functions as REST API calls, when that happens the ServiceModel files are going to get discarded. If you are redesigning this Openshift stuff it might be worth moving these functions into app/models and just expose these functions from the service models.